sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/ChatThreadsImpl.java

```diff
@@ -1959,6 +1868,151 @@ public ChatThreadProperties getChatThreadProperties(String chatThreadId) {
+     * @throws CommunicationErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> sendTypingNotificationAsync(
+            String chatThreadId, TypingNotificationOptions sendTypingNotificationRequest, Context context) {
+        return sendTypingNotificationWithResponseAsync(chatThreadId, sendTypingNotificationRequest, context)
+                .flatMap((Response<Void> res) -> Mono.empty());
+    }
+
+    /**
+     * Posts a typing event to a thread, on behalf of a user.
+     *
+     * @param chatThreadId Id of the thread.
+     * @param sendTypingNotificationRequest Details of the typing notification request.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws CommunicationErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public void sendTypingNotification(String chatThreadId, TypingNotificationOptions sendTypingNotificationRequest) {
+        sendTypingNotificationAsync(chatThreadId, sendTypingNotificationRequest).block();
+    }
+
+    /**
+     * Posts a typing event to a thread, on behalf of a user.
+     *
+     * @param chatThreadId Id of the thread.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws CommunicationErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public void sendTypingNotification(String chatThreadId) {
+        final TypingNotificationOptions sendTypingNotificationRequest = null;
+        sendTypingNotificationAsync(chatThreadId, sendTypingNotificationRequest).block();
+    }
+
+    /**
+     * Posts a typing event to a thread, on behalf of a user.
+     *
+     * @param chatThreadId Id of the thread.
+     * @param sendTypingNotificationRequest Details of the typing notification request.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws CommunicationErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the response.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<Void> sendTypingNotificationWithResponse(
+            String chatThreadId, TypingNotificationOptions sendTypingNotificationRequest, Context context) {
```

Comment:
options ?

Reply:
See other comment, this is the autogenerated client. It's the same for the previous renames too